app/src/main/java/com/andrew/tasky/agenda/data/reminder/ReminderRepositoryImpl.kt

```diff
@@ -1,18 +1,41 @@
 package com.andrew.tasky.agenda.data.reminder
 
+import android.app.Application
+import androidx.work.*
+import com.andrew.tasky.agenda.data.util.toReminderDto
+import com.andrew.tasky.agenda.data.util.toReminderEntity
 import com.andrew.tasky.agenda.domain.ReminderRepository
 import com.andrew.tasky.agenda.domain.models.AgendaItem
 import com.andrew.tasky.auth.data.AuthResult
+import com.google.gson.Gson
+import javax.inject.Inject
 
-class ReminderRepositoryImpl(
-    db: ReminderDatabase
+class ReminderRepositoryImpl @Inject constructor(
+    private val appContext: Application,
+    private val db: ReminderDatabase
 ) : ReminderRepository {
 
-    override suspend fun createReminder(reminderDto: ReminderDto): AuthResult<Unit> {
-        TODO("Not yet implemented")
+    override suspend fun createReminder(reminder: AgendaItem.Reminder) {
+        db.getReminderDao().upsert(reminder.toReminderEntity())
+
+        val reminderDtoAsJson = Gson().toJson(reminder.toReminderDto())
+        val uploadReminderRequest = OneTimeWorkRequestBuilder<UploadReminderWorker>()
+            .setConstraints(
+                Constraints.Builder()
+                    .setRequiredNetworkType(NetworkType.CONNECTED)
+                    .build()
+            ).setInputData(
+                workDataOf(
+                    "CREATE_REMINDER" to reminderDtoAsJson
+                )
+            )
+            .build()
+        WorkManager
+            .getInstance(appContext)
+            .enqueue(uploadReminderRequest)
```

Comment:
For creating reminders WorkManager shouldn't be necessary since it's not a longer thing, it's just one network call that takes maybe a second. WorkManager makes sense for example when uploading files (like photos for an event) where there's a higher chance that the user goes away or something goes wrong